processor/resourcedetectionprocessor/internal/resourcedetection.go

```diff
@@ -133,9 +143,10 @@ func (p *ResourceProvider) detectResource(ctx context.Context) {
 			mergedSchemaURL = MergeSchemaURL(mergedSchemaURL, schemaURL)
 			MergeResource(res, r, false)
 		}
-
 	}
 
+	FilterAttributes(res.Attributes(), p.attributesToKeep)
+
 	p.logger.Info("detected resource information", zap.Any("resource", AttributesToMap(res.Attributes())))
```

Comment:
Perhaps information about some attributes being filtered could be logged somehow (either here or using some debug log)?

Reply:
Are you thinkin log the attributes that were dropped?

Reply:
Yeah, that should be perfect

Reply:
Would you rather I:

A. Move the log on L150 before the filter and then create a new log entry after the filter that contains the updated list
B. Update the FilterAttribute method to return the list of dropped attributes (or nil) which could then be logged

Option A is simpler, but you'd have to do a comparison between the logs to see what was actually dropped.  Option B is more explicit with what was dropped, but adds in slight complexity and memory requirements

Reply:
Ended up going with option B as I think its the best customer experience.  Here is what it looks like:

<img width="660" alt="image" src="https://user-images.githubusercontent.com/12352919/159064512-e2e95c7e-8e8a-4457-8ff0-2328c4c288f9.png">

Reply:
Great, thank you! One small nit, I think the reverse logging order would be more natural to me (first "detected...", then "dropped..") but perhaps that's just me :)

Reply:
I opted for the "detected" to come after so that we could see exactly what is going to be applied after the filtering.  Maybe we could have a "detected" log always before filtering, and then if attributes were filtered we would have a "dropped" log and "finalized" log?  I think it is important to print out exactly what will be added in any scenario.

Reply:
If you mean keep the "detect" log as is after the filtering and then print out the "dropped" log, I like that too.